consensus/spos/consensusMessageValidator_test.go

```diff
@@ -1,3 +1,455 @@
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_InvalidLeaderSignatureSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("01"), AggregateSignature: sig, LeaderSignature: []byte("0")}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("01"), AggregateSignature: sig, LeaderSignature: sig}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithSignatureValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Body: []byte("body")}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithSignatureValidity_InvalidSignatureShareSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithSignatureValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{SignatureShare: sig}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidHeaderSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidHeaderSize))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Header: []byte("header")}
```

Comment:
You can use the same pattern used below for body size:

headerBytes := make([]byte, 100)
_, _ = rand.Read(headerBytes)
cnsMsg := &consensus.Message{Header: headerBytes}

Reply:
done.